File: ServeurCompteCourant/src/main/java/com/example/serveurcomptecourant/models/TypeTransaction.java
package com.example.serveurcomptecourant.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "types_transaction")
public class TypeTransaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id_type_transaction")
    private Integer id;

    @Column(name = "libelle", nullable = false, length = 50)
    private String libelle;

    @Column(name = "actif", nullable = false)
    private Boolean actif = true;

    @Column(name = "signe", nullable = false, length = 1)
    private String signe;

    // Constructors
    public TypeTransaction() {}

    public TypeTransaction(String libelle, Boolean actif, String signe) {
        this.libelle = libelle;
        this.actif = actif;
        this.signe = signe;
    }

    // Getters & Setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public Boolean getActif() {
        return actif;
    }

    public void setActif(Boolean actif) {
        this.actif = actif;
    }

    public String getSigne() {
        return signe;
    }

    public void setSigne(String signe) {
        this.signe = signe;
    }
}

File: ServeurCompteDepot/models/TypeTransaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ServeurCompteDepot.Models
{
    [Table("types_transaction")]
    public class TypeTransaction
    {
        [Key]
        [Column("id_type_transaction")]
        public int IdTypeTransaction { get; set; }

        [Required]
        [MaxLength(50)]
        [Column("libelle")]
        public string Libelle { get; set; } = string.Empty;

        [Required]
        [Column("actif")]
        public bool Actif { get; set; } = true;

        [Required]
        [MaxLength(1)]
        [Column("signe")]
        public string Signe { get; set; } = string.Empty;

        // Navigation properties
        public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
    }
}